azure-iot-common/azure/iot/common/asyncio_compat.py

```diff
@@ -0,0 +1,111 @@
+import asyncio
+import concurrent.futures
+
+
+def get_running_loop():
+    """
+    Helper function to use the best method to get an event loop to run async code
+    in.  This is a "best effort" function.  get_running_loop is preferred because it
+    works in more cases (especially if customers are using custom event loop policies),
+    but get_event_loop works well in simple cases (e.g. there is only one loop).
+    """
+    try:
+        loop = asyncio.get_running_loop()
+    except AttributeError:
+        loop = asyncio.get_event_loop()
+    return loop
+
+
+def create_task(coro):
+    """
+    "Best effort" function. If avaialable (Python 3.7+), use asyncio.create_task,
+    which is preferred as it is more specific for the goal of immediately scheduling
+    a task from a coroutine. If not avaialble, use the more general puprose 
+    asyncio.ensure_future
+    """
+    try:
+        task = asyncio.create_task(coro)
```

Comment:
>coro [](start = 35, length = 4)

Should this be given as coro()

Reply:
Good catch. Since I hadn't written tests yet (and this isn't used anywhere), this slipped by.

---
In reply to: [246957140](https://github.com/Azure/azure-iot-sdk-python-preview/pull/39#discussion_r246957140) [](ancestors = 246957140)

Reply:
I am not sure if this understanding is implicit or not. But for me I think it will be really clear if there is a description of the param coro saying that it is a coroutine call or name the param like coro_call or something along those lines.

---
In reply to: [246957640](https://github.com/Azure/azure-iot-sdk-python-preview/pull/39#discussion_r246957640) [](ancestors = 246957640,246957140)